server/src/test/java/org/apache/druid/client/ImmutableDruidDataSourceTest.java

```diff
@@ -26,39 +26,76 @@
 import com.google.common.collect.ImmutableSortedMap;
 import org.apache.druid.jackson.DefaultObjectMapper;
 import org.apache.druid.java.util.common.Intervals;
+import org.apache.druid.test.utils.TestUtils;
 import org.apache.druid.timeline.DataSegment;
 import org.apache.druid.timeline.DataSegment.PruneLoadSpecHolder;
-import org.junit.Assert;
 import org.junit.Test;
 
+import javax.annotation.Nonnull;
 import java.io.IOException;
 
 public class ImmutableDruidDataSourceTest
 {
   @Test
   public void testSerde() throws IOException
   {
-    final DataSegment segment = new DataSegment(
-        "test",
-        Intervals.of("2017/2018"),
-        "version",
-        null,
-        ImmutableList.of("dim1", "dim2"),
-        ImmutableList.of("met1", "met2"),
-        null,
-        1,
-        100L,
-        PruneLoadSpecHolder.DEFAULT
-    );
-    final ImmutableDruidDataSource dataSource = new ImmutableDruidDataSource(
-        "test",
-        ImmutableMap.of("prop1", "val1", "prop2", "val2"),
-        ImmutableSortedMap.of(segment.getId(), segment)
-    );
+    final DataSegment segment = getTestSegment();
+    final ImmutableDruidDataSource dataSource = getImmutableDruidDataSource(segment);
 
     final ObjectMapper objectMapper = new DefaultObjectMapper()
         .setInjectableValues(new Std().addValue(PruneLoadSpecHolder.class, PruneLoadSpecHolder.DEFAULT));
     final String json = objectMapper.writeValueAsString(dataSource);
-    Assert.assertEquals(dataSource, objectMapper.readValue(json, ImmutableDruidDataSource.class));
+
+    TestUtils.assertEqualsImmutableDruidDataSource(dataSource, objectMapper.readValue(json, ImmutableDruidDataSource.class));
+  }
+
+  @Test(expected = UnsupportedOperationException.class)
```

Comment:
Please use `ExpectedException` instead and check the exception message as well.

Reply:
Done.